imap/sequence.c

```diff
@@ -98,20 +98,19 @@ EXPORTED void seqset_add(struct seqset *seq, unsigned num, int ismember)
     if (ismember && num == seq->prev && seq->len && seq->set[seq->len-1].high == num)
         return;
 
-    if (num <= seq->prev)
-        fatal("numbers out of order", EX_SOFTWARE);
+    if (num > seq->prev) {
+        if (!ismember) {
+            seq->prev = num;
+            return;
+        }
 
-    if (!ismember) {
-        seq->prev = num;
-        return;
+        /* as if the previous number was given to us */
+        if (seq->flags & SEQ_SPARSE)
+            seq->prev = num - 1;
```

Comment:
I feel like we need to probably make seqset handle out of order sequences as a real thing rather than pulling magic tricks like this - but it's not like you've made it more evil than it already was!

Reply:
Yeah, the only evil that I added was to not fatal() if num < prev and to force a new set (RFC 5267 prefers 10,9,8 rather than 10:8).  Otherwise, the existing logic was left as-is.